src/engraving/dom/part.cpp

```diff
@@ -404,6 +406,48 @@ const InstrumentList& Part::instruments() const
     return _instruments;
 }
 
+const StringData* Part::stringData(const Fraction& tick) const
+{
+    if (!score()) {
+        return nullptr;
+    }
+
+    const StringTunings* stringTunings = nullptr;
+    const Instrument* instrument = this->instrument(tick);
+
+    for (Measure* measure = score()->firstMeasure(); measure; measure = measure->nextMeasure()) {
```

Comment:
This looks like a very inefficient implementation. This way of iterating over segments is quite slow. We probably need some cache to avoid it...